src/main/java/com/google/gcs/sdrs/controller/ServiceLayerException.java

```diff
@@ -0,0 +1,50 @@
+/*
```

Comment:
is this new class necessary ? can we just add the string instance variable to the extended base class and inject the appropriate message upon object instantiation as an argument?

Reply:
BaseController.errorResponse method relies on an exception type to generate proper error response. We can certainly use exception message to differentiate different type of errors. In this case, a new exception class works better as it'd not fundamentally how error response is generated